src/models/model.js

```diff
@@ -0,0 +1,67 @@
+const mongo = require('../mongo');
+const mongodbDriver = require('mongodb');
+const ObjectID = mongodbDriver.ObjectID;
+
+/**
+ * @typedef {Object} BaseModel
+ * @typedef {string|ObjectID} id - record id
+ */
+
+/**
+ * Base model
+ */
+class Model {
+  /**
+   * Model's collection
+   *
+   * @return {Collection}
+   */
+  static get collection() {
+    throw new Error('Collection method is not implemented');
+  }
+
+  /**
+   * Find record by query
+   * @param {object} query - query object
+   * @return {Promise<User>|null}
+   */
+  static async findOne(query) {
+    const searchResult = await this.collection.findOne(query);
+
+    if (!searchResult) return null;
+
+    return new this({
+      id: searchResult._id,
+      ...searchResult
+    });
+  }
+
+  /**
+   * Finds record by its id
+   * @param {BaseModel.id} id - record's id
```

Comment:
```suggestion
   * @param {String} id - record's id
```
а то IDE не подхватывает такой синтаксис